src/Notepads/Controls/TextEditor/TextEditorCore.cs

```diff
@@ -825,6 +846,136 @@ public async void SearchInWeb()
             }
         }
 
+        private void AddIndentation()
+        {
+            GetTextSelectionPosition(out var start, out var end);
+            GetLineColumnSelection(out var startLine, out var endLine, out var startColumn, out var endColumn, out _, out _);
+
+            var startLineInitialIndex = start - startColumn + 1;
+            var endLineFinalIndex = end - endColumn + _contentLinesCache[endLine - 1].Length + 1;
+            if (endLineFinalIndex > _content.Length) endLineFinalIndex = _content.Length;
+
+            var tabStr = EditorSettingsService.EditorDefaultTabIndents == -1
+                ? "\t"
+                : new string(' ', EditorSettingsService.EditorDefaultTabIndents);
+            
+            // Handle single line selection scenario where part of the line is selected
+            if (start == end ||
```

Comment:
I found an edge case, changing it to below will fix the issue:

            if (start == end || 
                startColumn != 1 ||
                (startLine == endLine && end - start < _contentLinesCache[startLine - 1].Length))

![image](https://user-images.githubusercontent.com/1166162/77106443-518ae480-69dc-11ea-9de1-8e6a3900d3d1.png)

Reply:
Can't open the image.

Reply:
I think it would be better if when only one line is selected instead of indenting tab is inserted. Since user can insert tab for one line by clicking at the beginning. More like this:
 
 ```
         if (startLine == endLine)
         {
             Document.Selection.TypeText(tabStr);
             Document.Selection.StartPosition = Document.Selection.EndPosition;
             return;
         }
 ```

Reply:
This should work:

        if (startLine == endLine)
        {
            if (!(startColumn == 1 && endColumn >= _contentLinesCache[startLine - 1].Length))
            {
                Document.Selection.TypeText(tabStr);
                Document.Selection.StartPosition = Document.Selection.EndPosition;
                return;
            }
        }

You can verify as well. Basically selecting from column index 2 all the way to the end of the line including the line ending.

Reply:
I agree, this is just to copy VS Code behavior. Not 100% necessary. Let's just make it 

 if (startLine == endLine) 

for simplicity here.